doc/contributing.rst

```diff
@@ -0,0 +1,874 @@
+document your fix, enhancement or (unavoidable) breaking change.  Make sure to include the
+GitHub issue number when adding your entry (using `` :issue:`1234` `` where `1234` is the
+issue/pull request number).
+
+If your code is an enhancement, it is most likely necessary to add usage
+examples to the existing documentation.  This can be done following the section
+regarding documentation :ref:`above <contributing.documentation>`.
+
+Contributing your changes to *xarray*
+=====================================
+
+Committing your code
+--------------------
+
+Keep style fixes to a separate commit to make your pull request more readable.
+
+Once you've made changes, you can see them by typing::
+
+    git status
+
+If you have created a new file, it is not being tracked by git. Add it by typing::
+
+    git add path/to/file-to-be-added.py
+
+Doing 'git status' again should give something like::
+
+    # On branch shiny-new-feature
+    #
+    #       modified:   /relative/path/to/file-you-added.py
+    #
+
+Finally, commit your changes to your local repository with an explanatory message.
+*Xarray* uses a convention for commit message prefixes and layout.  Here are
+some common prefixes along with general guidelines for when to use them:
+
+    * ENH: Enhancement, new functionality
+    * BUG: Bug fix
+    * DOC: Additions/updates to documentation
+    * TST: Additions/updates to tests
+    * BLD: Updates to the build process/scripts
+    * PERF: Performance improvement
+    * CLN: Code cleanup
+
+The following defines how a commit message should be structured.  Please reference the
+relevant GitHub issues in your commit message using GH1234 or #1234.  Either style
+is fine, but the former is generally preferred:
+
+    * a subject line with `< 80` chars.
+    * One blank line.
+    * Optionally, a commit message body.
+
+Now you can commit your changes in your local repository::
+
+    git commit -m
+
+Combining commits
+-----------------
+
+If you have multiple commits, you may want to combine them into one commit, often
+referred to as "squashing" or "rebasing".  This is a common request by package maintainers
+when submitting a pull request as it maintains a more compact commit history.  To rebase
+your commits::
+
+    git rebase -i HEAD~#
+
+Where # is the number of commits you want to combine.  Then you can pick the relevant
+commit message and discard others.
+
+To squash to the master branch do::
+
+    git rebase -i master
+
+Use the ``s`` option on a commit to ``squash``, meaning to keep the commit messages,
+or ``f`` to ``fixup``, meaning to merge the commit messages.
+
+Then you will need to push the branch (see below) forcefully to replace the current
+commits with the new ones::
+
+    git push origin shiny-new-feature -f
+
+.. note::
+
+   GitHub now supports squashing commits when merging pull requests. Unless
+   there is a specific reason for squashing/rebasing your feature branch, we
+   recommend letting Github do the work.
```

Comment:
Given this note, is the section less relevant? Would have been v useful prior to the GH change

Reply:
I moved the `.. Note` to the top of this section. I think having some instructions on how/why to rebase is still useful so I've left the majority of this section in tact.

Reply:
My preference would be to not encourage rebasing. In my experience, it can be a source of confusion for many new contributor. Also, from a review perspective GitHub makes it easier to see incremental changes if you only append commits. A rebase in a large PR means I have to start review over from scratch.